zerver/tests/test_markdown_thumbnail.py

```diff
@@ -334,3 +343,42 @@ def test_thumbnail_multiple_messages(self) -> None:
             private_message_id,
             f'<p>This <a href="/user_uploads/{path_id}">image</a> is private</p>\n{rendered_thumb}',
         )
+
+    def test_thumbnail_race(self) -> None:
+        """Test what happens when thumbnailing happens between rendering and sending"""
+        path_id = self.upload_image("img.png")
+
+        self.assert_length(ImageAttachment.objects.get(path_id=path_id).thumbnail_metadata, 0)
+
+        # Render, but do not send, the message referencing the image.
+        # This will render as a spinner, since the thumbnail has not
+        # been generated yet.
+        message = check_message(
```

Comment:
`message` here, not being a Message object, should be called `send_request`; I'll make the tweak and merge.

Reply:
Diff is:
``` diff
diff --git a/zerver/tests/test_markdown_thumbnail.py b/zerver/tests/test_markdown_thumbnail.py
index f80416e195..59ca8e1d26 100644
--- a/zerver/tests/test_markdown_thumbnail.py
+++ b/zerver/tests/test_markdown_thumbnail.py
@@ -353,7 +353,7 @@ class MarkdownThumbnailTest(ZulipTestCase):
         # Render, but do not send, the message referencing the image.
         # This will render as a spinner, since the thumbnail has not
         # been generated yet.
-        message = check_message(
+        send_request = check_message(
             self.example_user("othello"),
             Client.objects.get_or_create(name="test suite")[0],
             Addressee.for_stream_name("Verona", "test"),
@@ -364,7 +364,7 @@ class MarkdownThumbnailTest(ZulipTestCase):
             f'<div class="message_inline_image"><a href="/user_uploads/{path_id}" title="image">'
             '<img class="image-loading-placeholder" src="/static/images/loading/loader-black.svg"></a></div>'
         )
-        self.assertEqual(message.message.rendered_content, expected)
+        self.assertEqual(send_request.message.rendered_content, expected)
 
         # Thumbnail the image.  The message does not exist yet, so
         # nothing is re-written.
@@ -373,7 +373,7 @@ class MarkdownThumbnailTest(ZulipTestCase):
         # Send the message; this should re-check the ImageAttachment
         # data, find the thumbnails, and update the rendered_content
         # to no longer contain a spinner.
-        message_id = do_send_messages([message])[0].message_id
+        message_id = do_send_messages([send_request])[0].message_id
 
         rendered_thumb = (
             f'<div class="message_inline_image"><a href="/user_uploads/{path_id}" title="image">'
```